Cesium3DTilesSelection/test/TestPntsToGltfConverter.cpp

```diff
@@ -0,0 +1,1275 @@
+#include "ConvertTileToGltf.h"
+
+#include <CesiumAsync/AsyncSystem.h>
+#include <CesiumAsync/HttpHeaders.h>
+#include <CesiumGltf/ExtensionCesiumRTC.h>
+#include <CesiumGltf/ExtensionKhrMaterialsUnlit.h>
+#include <CesiumGltf/ExtensionMeshPrimitiveExtFeatureMetadata.h>
+#include <CesiumGltf/ExtensionModelExtFeatureMetadata.h>
+#include <CesiumUtility/Math.h>
+
+#include <catch2/catch.hpp>
+#include <rapidjson/document.h>
+#include <spdlog/sinks/ringbuffer_sink.h>
+#include <spdlog/spdlog.h>
+
+#include <filesystem>
+#include <set>
+
+using namespace CesiumGltf;
+using namespace Cesium3DTilesSelection;
+using namespace CesiumUtility;
+
+template <typename Type>
+static void checkBufferContents(
+    const std::vector<std::byte>& buffer,
+    const std::vector<Type>& expected) {
+  REQUIRE(buffer.size() == expected.size() * sizeof(Type));
+  const int32_t byteStride = sizeof(Type);
+  if constexpr (std::is_same_v<Type, glm::vec3>) {
+    for (size_t i = 0; i < expected.size(); ++i) {
+      const glm::vec3& value =
+          *reinterpret_cast<const glm::vec3*>(buffer.data() + i * byteStride);
+      const glm::vec3& expectedValue = expected[i];
+      REQUIRE(Math::equalsEpsilon(
+          static_cast<glm::dvec3>(value),
+          static_cast<glm::dvec3>(expectedValue),
+          Math::Epsilon6));
+    }
+  } else if constexpr (std::is_same_v<Type, glm::vec4>) {
+    for (size_t i = 0; i < expected.size(); ++i) {
+      const glm::vec4& value =
+          *reinterpret_cast<const glm::vec4*>(buffer.data() + i * byteStride);
+      const glm::vec4& expectedValue = expected[i];
+      REQUIRE(Math::equalsEpsilon(
+          static_cast<glm::dvec4>(value),
+          static_cast<glm::dvec4>(expectedValue),
+          Math::Epsilon6));
+    }
+  } else if constexpr (std::is_floating_point_v<Type>) {
+    for (size_t i = 0; i < expected.size(); ++i) {
+      const Type& value =
+          *reinterpret_cast<const Type*>(buffer.data() + i * byteStride);
+      const Type& expectedValue = expected[i];
+      REQUIRE(value == Approx(expectedValue));
+    }
+  } else if constexpr (
+      std::is_integral_v<Type> || std::is_same_v<Type, glm::u8vec2> ||
+      std::is_same_v<Type, glm::u8vec3> || std::is_same_v<Type, glm::u8vec4>) {
+    for (size_t i = 0; i < expected.size(); ++i) {
+      const Type& value =
+          *reinterpret_cast<const Type*>(buffer.data() + i * byteStride);
+      const Type& expectedValue = expected[i];
+      REQUIRE(value == expectedValue);
+    }
+  } else {
+    FAIL("Buffer check has not been implemented for the given type.");
+  }
+}
+
+template <typename Type>
+static void checkBufferContents(
```

Comment:
You can add an attribute to the parameter:
https://en.cppreference.com/w/cpp/language/attributes/maybe_unused
You can also use specialization rather than if constexpr, but that's a bigger change.